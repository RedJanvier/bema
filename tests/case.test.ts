import './__setup'
import { constant, noop } from 'lodash'
import { stripKey } from './__helpers'

describe('validation', () => {
  test.skip('If receives unknown parameters then an error is thrown', () => {
    const b = bema as any
    b.case({ name: '1' }).run(noop)
    expect(run(b)).rejects.toMatchInlineSnapshot(
      `"Invalid parameter values given. The following parameter types do not exist: name. There are no parameters types defined."`
    )
  })
})

test('before is called once before the case. It receives contextual data.', async () => {
  const ga = bema.group('a')
  ga.case().run(constant(1)).after(f1)
  ga.case().run(constant(2)).after(f1)
  ga.case().run(constant(3)).after(f1)
  await run(bema)
  expect(f1.mock.calls.length).toEqual(3)
  expect(stripKey('stats', f1.mock.calls)).toMatchSnapshot()
})

test('after is called once after the case. It receives info about return value and result report.', async () => {
  const ga = bema.group('a')
  ga.case().run(constant(1)).after(f1)
  ga.case().run(constant(2)).after(f1)
  ga.case().run(constant(3)).after(f1)
  await run(bema)
  expect(f1.mock.calls.length).toEqual(3)
  expect(stripKey('stats', f1.mock.calls)).toMatchSnapshot()
})
