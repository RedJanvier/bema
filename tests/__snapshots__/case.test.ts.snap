// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`after is called once after the case. It receives info about return value and result report. 1`] = `
Object {
  "0": Array [
    Object {
      "$info": Object {
        "report": Object {
          "group": "a",
          "matrix": Array [],
          "name": "<Anonymous>",
          "parameters": Object {},
          "parametersFromMatrix": Object {},
          "parametersFromNotMatrix": Object {},
          "quickMode": true,
          "stats": "STRIPPED",
        },
        "returnValue": 1,
      },
    },
  ],
  "1": Array [
    Object {
      "$info": Object {
        "report": Object {
          "group": "a",
          "matrix": Array [],
          "name": "<Anonymous>",
          "parameters": Object {},
          "parametersFromMatrix": Object {},
          "parametersFromNotMatrix": Object {},
          "quickMode": true,
          "stats": "STRIPPED",
        },
        "returnValue": 2,
      },
    },
  ],
  "2": Array [
    Object {
      "$info": Object {
        "report": Object {
          "group": "a",
          "matrix": Array [],
          "name": "<Anonymous>",
          "parameters": Object {},
          "parametersFromMatrix": Object {},
          "parametersFromNotMatrix": Object {},
          "quickMode": true,
          "stats": "STRIPPED",
        },
        "returnValue": 3,
      },
    },
  ],
}
`;

exports[`before is called once before the case. It receives contextual data. 1`] = `
Object {
  "0": Array [
    Object {
      "$info": Object {
        "report": Object {
          "group": "a",
          "matrix": Array [],
          "name": "<Anonymous>",
          "parameters": Object {},
          "parametersFromMatrix": Object {},
          "parametersFromNotMatrix": Object {},
          "quickMode": true,
          "stats": "STRIPPED",
        },
        "returnValue": 1,
      },
    },
  ],
  "1": Array [
    Object {
      "$info": Object {
        "report": Object {
          "group": "a",
          "matrix": Array [],
          "name": "<Anonymous>",
          "parameters": Object {},
          "parametersFromMatrix": Object {},
          "parametersFromNotMatrix": Object {},
          "quickMode": true,
          "stats": "STRIPPED",
        },
        "returnValue": 2,
      },
    },
  ],
  "2": Array [
    Object {
      "$info": Object {
        "report": Object {
          "group": "a",
          "matrix": Array [],
          "name": "<Anonymous>",
          "parameters": Object {},
          "parametersFromMatrix": Object {},
          "parametersFromNotMatrix": Object {},
          "quickMode": true,
          "stats": "STRIPPED",
        },
        "returnValue": 3,
      },
    },
  ],
}
`;
